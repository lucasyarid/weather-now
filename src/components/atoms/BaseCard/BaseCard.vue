<template>
  <div class="BaseCard">
    <div v-if="title" class="BaseCard__title">
      {{ title }}
    </div>
    <div class="BaseCard__content">
      <slot />
    </div>
    <div class="BaseCard__footer">
      <slot name="footer" />
      <div class="BaseCard__bottom-text">{{ bottomText }}</div>
    </div>
  </div>
</template>

<script>
export default {
  name: "BaseCard",
  props: {
    title: {
      type: String,
      default: ""
    },
    bottomText: {
      type: String,
      default: ""
    }
  }
};
</script>

<style lang="scss" scoped>
.BaseCard {
  box-shadow: 0px 0px 5px 0px $color-box-shadow;
  margin: 40px;
  background-color: white;
  border-radius: 5px;
  height: fit-content;

  @media (min-width: 760px) {
    margin: 20px;
    min-width: 300px;
  }

  &__title,
  &__content,
  &__footer {
    padding: 15px;
    text-align: center;
    color: $color-darker;
  }
  &__title {
    border-bottom: 1px solid $color-light;
  }
  &__footer {
    background-color: $color-xlight;
  }
  &__bottom-text {
    margin-top: 10px;
    font-size: 10px;
  }
}
</style>
