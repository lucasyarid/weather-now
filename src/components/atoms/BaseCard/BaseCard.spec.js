import { shallowMount } from "@vue/test-utils";
import BaseCard from "./BaseCard";

describe("BaseCard", () => {
  const WRAPPER = shallowMount(BaseCard);

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
