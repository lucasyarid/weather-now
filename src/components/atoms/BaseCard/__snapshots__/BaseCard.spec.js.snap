// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard Componente n√£o alterado 1`] = `"<div class=\\"BaseCard\\"><!----> <div class=\\"BaseCard__content\\"></div> <div class=\\"BaseCard__footer\\"> <div class=\\"BaseCard__bottom-text\\"></div></div></div>"`;
